raster-foundry/raster-foundry
# Add secondary data model authorization

## Overview

This PR implements authorization on endpoints according to requirements in #3368.
 
### Checklist

- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated (raster-foundry/raster-foundry-api-spec#16)

### Notes

This is a retarget of #3417 to develop instead of to the platforms branch, which is now merged.

`MapTokenDao` new sql strings aren't tested, but everything else is. The choice to exclude them was
based on not having any of the structure for tool runs/analyses in the db testing code yet. The only
place that code would be exercised is sharing analyses.

## Testing Instructions

 * CI
 * inspect logic for `authorize` methods on AOI and MapToken and usage in both of their routes
 * compare authorization against checklist in #3368 
 * create an analysis, attempt to share a node, and confirm that you don't get any errors from the API

Checks a lot of boxes in #3368
